### PR TITLE
Fix deprecation issue #375 and more

### LINKED ISSUE #375: django.conf.urls.url() is deprecated in favor of django.urls.re_path()
```
todolists/tests/test_views.py::test_create_todolist
  /home/jelle/projects/archweb/urls.py:84: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
    url(r'^groups/', include(packages.urls_groups)),

todolists/tests/test_views.py::test_create_todolist
  /home/jelle/projects/archweb/urls.py:85: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
    url(r'^mirrorlist/', include(mirrors.urls_mirrorlist)),

todolists/tests/test_views.py::test_create_todolist
  /home/jelle/projects/archweb/urls.py:86: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
    url(r'^mirrors/', include(mirrors.urls)),

todolists/tests/test_views.py::test_create_todolist
  /home/jelle/projects/archweb/urls.py:87: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
    url(r'^news/', include(news.urls)),

todolists/tests/test_views.py::test_create_todolist
  /home/jelle/projects/archweb/urls.py:88: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
    url(r'^packages/', include(packages.urls)),

todolists/tests/test_views.py::test_create_todolist
  /home/jelle/projects/archweb/urls.py:89: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
    url(r'^releng/', include(releng.urls)),

todolists/tests/test_views.py::test_create_todolist
  /home/jelle/projects/archweb/urls.py:90: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
    url(r'^todo/', include(todolists.urls)),

todolists/tests/test_views.py::test_create_todolist
  /home/jelle/projects/archweb/urls.py:91: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
    url(r'^visualize/', include(visualize.urls)),

todolists/tests/test_views.py::test_create_todolist
  /home/jelle/projects/archweb/urls.py:92: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
    url(r'^opensearch/packages/$', packages.views.opensearch, name='opensearch-packages'),

todolists/tests/test_views.py::test_create_todolist
  /home/jelle/projects/archweb/urls.py:93: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
    url(r'^opensearch/packages/suggest$', packages.views.opensearch_suggest, name='opensearch-packages-suggest'),

todolists/tests/test_views.py::test_create_todolist
  /home/jelle/projects/archweb/urls.py:98: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
    url(r'^sitemap.xml$', cache_page(1831)(sitemap_views.index),

todolists/tests/test_views.py::test_create_todolist
  /home/jelle/projects/archweb/urls.py:100: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
    url(r'^sitemap-(?P<section>.+)\.xml$', cache_page(1831)(sitemap_views.sitemap),

todolists/tests/test_views.py::test_create_todolist
  /home/jelle/projects/archweb/urls.py:103: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
    url(r'^news-sitemap\.xml$', cache_page(1831)(sitemap_views.sitemap),

todolists/tests/test_views.py::test_create_todolist
  /home/jelle/projects/archweb/urls.py:110: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
    url(r'^login/$', auth_views.LoginView.as_view(template_name='registration/login.html'), name='login'),

todolists/tests/test_views.py::test_create_todolist
  /home/jelle/projects/archweb/urls.py:111: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
    url(r'^logout/$', auth_views.LogoutView.as_view(template_name='registration/logout.html'), name='logout'),
```

### PR DESCRIPTION
**Pull request by AbhishekSrikanth**
- Replaced `url` with `re_path`. Fixes #375 
- Renamed requirement_tests.py to requirements_test.py to match README.

### CODE REVIEW

**Comment by jelly** [comment]
Please resolve the flake8 warnings :)

Another improvement would be to not replace everything with `re_path` as `path` works fine for routes such as https://github.com/archlinux/archweb/pull/380/files#diff-5a824835ab37f3a09e03cee50b3dd3b8a4579d61ec760032073a5d162df7acceR44

**Review by jelly — CHANGES_REQUESTED** [review]
Please resolve the failing CI

File: urls.py [hunk 1]
```diff
@@ -41,74 +40,74 @@
...
+    re_path(r'^groups/', include(packages.urls_groups)),
+    re_path(r'^mirrorlist/', include(mirrors.urls_mirrorlist)),
+    re_path(r'^mirrors/', include(mirrors.urls)),
+    re_path(r'^news/', include(news.urls)),
+    re_path(r'^packages/', include(packages.urls)),
+    re_path(r'^releng/', include(releng.urls)),
+    re_path(r'^todo/', include(todolists.urls)),
+    re_path(r'^visualize/', include(visualize.urls)),
+    re_path(r'^opensearch/packages/$', packages.views.opensearch, name='opensearch-packages'),
+    re_path(r'^opensearch/packages/suggest$', packages.views.opensearch_suggest, name='opensearch-packages-suggest'),
```
Comment (jelly) [inline urls.py:92]: most of these urls can be converted to a simple `path('admin/') for example.

**Review by jelly — CHANGES_REQUESTED** [review]

File: urls.py (hunk 1 above)
Comment (AbhishekSrikanth) [reply urls.py:92]: Hi @jelly , yes using `path` for such scenarios makes sense. Getting those changes ready. Thanks!.

**Review by jelly — APPROVED** [review]

**Comment by jelly** [comment]
Thanks!

**Pull request merged**